### ISSUE TITLE
PHP Warning – yii\base\ErrorException Expected array for frame 0

### ISSUE THREAD

**Issue by mrsombre** [yiisoft/yii2#11120]
Problem with ErrorHandler. 
Confirmed for PHP 7.0.4, yii 2.0.7, CLI (\yii\console\Application), YII_DEBUG enabled, xdebug enabled.

Reproducing:
- cause any warning inside console controller
```
$a = [];
$b = $a['ntf'];
```
- call CLI action ./yii controller/action with warning
- you'll see stack of errors

```
PHP Warning:  Expected array for frame 0 in /.../vendor/yiisoft/yii2/console/ErrorHandler.php on line 49
PHP Stack trace:
PHP   1. yii\base\ErrorHandler->handleException() /.../vendor/yiisoft/yii2/base/ErrorHandler.php:0
PHP   2. Exception->__toString() /.../vendor/yiisoft/yii2/base/ErrorHandler.php:120
PHP   3. Exception->getTraceAsString() /.../vendor/yiisoft/yii2/base/ErrorHandler.php:120
PHP   4. yii\base\ErrorHandler->handleError() /.../vendor/yiisoft/yii2/base/ErrorHandler.php:120
PHP   5. yii\base\ErrorHandler->handleException() /.../vendor/yiisoft/yii2/base/ErrorHandler.php:203
PHP   6. yii\console\ErrorHandler->renderException() /.../vendor/yiisoft/yii2/base/ErrorHandler.php:109
PHP   7. Exception->getTraceAsString() /.../vendor/yiisoft/yii2/console/ErrorHandler.php:49
PHP Warning 'yii\base\ErrorException' with message 'Expected array for frame 0'

in /.../vendor/yiisoft/yii2/base/ErrorHandler.php:120
```

Switched to custom ErrorHandler using ErrorException without xdebug related code fixes problem.
Look at \yii\base\ErrorException:45

**Comment by SilverFire** [comment]
It's a heisenbug :fearful: 

I failed to reproduce it in the described way, however I spectate this error in my project on local env sometimes. Usually it happens just after system load and I fix it in non-obvious way like page refresh or disabling/enabling X-Debug listener in PHPStorm. I have no idea why does it happen :(

**Comment by hiqsol** [comment]
I receive this error too. With other code. Console application too.
PHP 7.0 only, PHP 5.6 ok.

**Comment by cebe** [comment]
> I failed to reproduce it in the described way, however I spectate this error in my project on local env sometimes.

same here...

**Comment by Koudy** [comment]
+

**Comment by SilverFire** [comment]
If anybody have confirmed way to reproduce the error - please, share it!

**Comment by quantum13** [comment]
Php 7.0.0, xdebug enabled:
for web app
in index.php:     defined('YII_ENV_TEST') or define('YII_ENV_TEST', true); (to pass in branch with output as string).

in controller, before class
```php
interface A {public static function a();}
class B implements A{};
```

in action
```php
Yii::$app->response->format = Response::FORMAT_HTML;
new B();
```

get this action and you will see error `PHP Warning 'yii\base\ErrorException' with message 'Expected array for frame 0`.

after deleting $exception->getTraceAsString() in
https://github.com/yiisoft/yii2/blob/master/framework/base/ErrorHandler.php#L321
you will see normal exception

**Comment by akaNightmare** [comment]
+1

**Comment by gbespyatykh** [comment]
+1

Same situation in console and web application on ErrorException

PHP 7.0.5 + xdebug 2.4.1-dev + Yii 2.0.7

**Comment by wallysalami** [comment]
Just to reinforce the problem: we are seeing this `PHP Warning:  Expected array for frame 0` in the web application sometimes, but only in computers with xdebug enabled.

One example happened in a AJAX call. One of our classes was extending from `Model`, but the developer has forgot to include `use yii\base\Model`. In her Linux machine with xdebug *enabled*, that generic warning showed up *without the full stack trace*. In my Mac with xdebug *disabled*, I've got the warning `Class not found` *with the full stack trace*.

**Comment by Hector68** [comment]
You can delete files in ```runtime```

**Comment by junthink** [comment]
+1

**Comment by igoryan-909** [comment]
+1

**Comment by slinstj** [comment]
A simple `die('anything');` inside any action will cause this error.
This is the my [stack trace](https://gist.github.com/slinstj/b208ef0f4e553906b5919a686e54ad9c).

(PHP 7, xdebug enabled, Linux)

**Comment by slinstj** [comment]
For a while, I disabled Yii debug in `main-local.php` so everything seems to work ok again.

**Comment by luiseduardo** [comment]
+1

**Comment by jolinp** [comment]
+1

**Comment by jolinp** [comment]
I tried the following steps, and the issue is disappeared :
  1. set `xdebug.remote_enable=false` in `ext-xdebug.ini`, and restart php-fpm. 
  2. reset `xdebug.remote_enable=true` and restart php-fpm again. 

(PHP 7.05, xdebug 2.4.0, PhpStorm)

**Comment by SilverFire** [comment]
@jolinp sometimes even fpm restart solves the problem

**Comment by astappev** [comment]
10 minutes ago, I had the same warning. I tried to disable xdebug and it did not help me. Then I switched to php 5.6.16 and I god error:
`Allowed memory size of xxx bytes exhausted (tried to allocate yyy bytes) in php`
(I tried to create preview of big image).

I increased memory limit and it solve the problem with memory. Then I switched back to php 7.0.2 and anything is OK.

Maybe it will be helpful.

**Comment by samdark** [comment]
@astappev when you've switched back did you increase PHP 7 allowed memory as well?

**Comment by astappev** [comment]
Yes, of course. I think it can be partially related to that

**Comment by uran1980** [comment]
Hi, I had the same error.

My env is:
* Kubuntu 16.04
* php 7.0.4-7ubuntu2.1
* Yii2 2.0.8

When I set **chmod 777 -R** on ```.../{backend|frontend}/runtime``` the problem was solved.

**Comment by slinstj** [comment]
Clear everything inside `.../runtime` worked to me (at least for now). It was already 777.

My env is PHP 7.0.4-7ubuntu2.1 / xdebug v2.4.0

**Comment by tanakahisateru** [comment]
I could reproduce similar behavior using below script:

```php
<?php
require __DIR__ . '/vendor/autoload.php';
use yii\base\ErrorException;

function handleError($code, $message, $file, $line) {
    throw new ErrorException($message, $code, $code, $file, $line);
}
set_error_handler('handleError');

class Klazz {
    public static function foo() {
        static::bar();
    }
    private static function bar() {
        eval("echo \$foo['bar'];");
    }
}

Klazz::foo();
```

```
$ php -v
PHP 7.0.7 (cli) (built: May 25 2016 17:35:28) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies
    with Xdebug v2.4.0, Copyright (c) 2002-2016, by Derick Rethans

$ php xdebugtest.php 
PHP Fatal error:  Uncaught yii\base\ErrorException: Expected array for frame 0 in Unknown:0
Stack trace:
#0 {main}
  thrown in Unknown on line 0
```

Just when `use yii\base\ErrorException;` line removed, I could get expected result:

```
$ php xdebugtest.php 
PHP Fatal error:  Uncaught ErrorException: Undefined variable: foo in /.../xdebugtest.php(22) : eval()'d code:1
Stack trace:
#0 /.../xdebugtest.php(22) : eval()'d code(1): handleError(8, 'Undefined varia...', '/...', 1, Array)
#1 /.../xdebugtest.php(22): eval()
#2 /.../xdebugtest.php(17): Klazz::bar()
#3 /.../xdebugtest.php(26): Klazz::foo()
#4 {main}
  thrown in /.../xdebugtest.php(22) : eval()'d code on line 1
```

**Comment by tanakahisateru** [comment]
Isn't it closed?

**Comment by cebe** [comment]
yes, thanks!

**Comment by 86868057** [comment]
this problem may stil reproduce by Yii2
the flowing codes are not included any phpfile and it doesn't throw error
```php
class MyException extends \ErrorException{
	public function __toString(){
		return parent::__toString();
	}
}
$e = new MyException('abc');
echo $e->__toString();
```

---

but it's happened when running in Yii2 controller
```php
namespace app\controllers;

class MyException extends \yii\base\ErrorException{
	public function __toString(){
		return parent::__toString();
	}
}
/**
 * Site controller
 */
class SiteController extends \yii\web\Controller
{
    public function actionMain(){
		throw new MyException('abc');
    }
}
```

**Comment by cebe** [comment]
the fix will be released with 2.0.9, it is expected to not work with 2.0.8

**Comment by hasalazar** [comment]
This problem can be a cause of having the name of a function repeated in the model.

This happened to me and solve it by changing the name to the function

`PHP Warning – yii\base\ErrorException
Expected array for frame 0`
